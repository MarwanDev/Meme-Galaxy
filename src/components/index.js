export { default as Navbar } from './Navbar/Navbar';
export { default as Header } from './Header/Header';
export { default as Categories } from './Categories/Categories';
export { default as Layout } from './Layout/Layout';
export { default as Category } from './Category/Category';
