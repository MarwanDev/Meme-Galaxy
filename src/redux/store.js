import { configureStore } from "@reduxjs/toolkit";
import memesReducer from "./memesSlice/memesSlice";

const store = configureStore({
  reducer: {
    meme: memesReducer,
  },
});

export default store;
