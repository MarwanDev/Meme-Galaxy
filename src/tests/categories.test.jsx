import { Categories } from "../components";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

describe("Categories", () => {
  it("renders categories correctly", () => {
    const categories = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Categories />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(categories).toMatchSnapshot();
  });
  it("renders categories component correctly", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Categories />
        </Provider>
      </BrowserRouter>
    );
  });
});
