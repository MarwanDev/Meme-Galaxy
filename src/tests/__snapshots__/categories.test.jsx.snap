// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Categories > renders categories correctly 1`] = `
<div
  className="categories-container"
>
  <ul
    style={
      {
        "display": "flex",
        "flexWrap": "wrap",
        "gap": 40,
        "justifyContent": "space-between",
        "listStyleType": "none",
        "paddingTop": 10,
      }
    }
  >
    <li>
      <div
        className="category-container"
        href=""
        id={2}
        onClick={[Function]}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "textAlign": "end",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
        <a
          className=""
          href="/details"
          id={2}
          onClick={[Function]}
          style={
            {
              "fontSize": 60,
              "textDecoration": "none",
            }
          }
        >
          ðŸ¤£
        </a>
        <h6
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "marginTop": 4,
              "textAlign": "right",
              "textDecoration": "none",
            }
          }
        >
          Mr Bala7
        </h6>
      </div>
    </li>
    <li>
      <div
        className="category-container"
        href=""
        id={3}
        onClick={[Function]}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "textAlign": "end",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
        <a
          className=""
          href="/details"
          id={3}
          onClick={[Function]}
          style={
            {
              "fontSize": 60,
              "textDecoration": "none",
            }
          }
        >
          ðŸ¤¡
        </a>
        <h6
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "marginTop": 4,
              "textAlign": "right",
              "textDecoration": "none",
            }
          }
        >
          Mr Betengana
        </h6>
      </div>
    </li>
    <li>
      <div
        className="category-container"
        href=""
        id={4}
        onClick={[Function]}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "textAlign": "end",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
        <a
          className=""
          href="/details"
          id={4}
          onClick={[Function]}
          style={
            {
              "fontSize": 60,
              "textDecoration": "none",
            }
          }
        >
          ðŸ˜…
        </a>
        <h6
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "marginTop": 4,
              "textAlign": "right",
              "textDecoration": "none",
            }
          }
        >
          Mr Basterma
        </h6>
      </div>
    </li>
    <li>
      <div
        className="category-container"
        href=""
        id={5}
        onClick={[Function]}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "textAlign": "end",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
        <a
          className=""
          href="/details"
          id={5}
          onClick={[Function]}
          style={
            {
              "fontSize": 60,
              "textDecoration": "none",
            }
          }
        >
          ðŸ˜Ž
        </a>
        <h6
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "marginTop": 4,
              "textAlign": "right",
              "textDecoration": "none",
            }
          }
        >
          Mrs Bala7
        </h6>
      </div>
    </li>
    <li>
      <div
        className="category-container"
        href=""
        id={0}
        onClick={[Function]}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "textAlign": "end",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
        <a
          className=""
          href="/details"
          id={0}
          onClick={[Function]}
          style={
            {
              "fontSize": 60,
              "textDecoration": "none",
            }
          }
        >
          ðŸ¤‘  
        </a>
        <h6
          style={
            {
              "alignSelf": "flex-end",
              "color": "#fff",
              "marginTop": 4,
              "textAlign": "right",
              "textDecoration": "none",
            }
          }
        >
          All
        </h6>
      </div>
    </li>
  </ul>
</div>
`;
